src/web/app/components/comment-box/comment-table-modal/comment-table-modal.component.ts

```diff
@@ -0,0 +1,232 @@
+  showResponsesToInCommentVisibilityType: CommentVisibilityType[] = [];
+
+  constructor(private commentService: FeedbackResponseCommentService,
+              public activeModal: NgbActiveModal) { }
+
+  ngOnInit(): void {
+    this.showResponsesToInCommentVisibilityType =
+      this.showResponsesTo.map((visibilityType: FeedbackVisibilityType) => {
+        switch (visibilityType) {
+          case FeedbackVisibilityType.INSTRUCTORS:
+            return CommentVisibilityType.INSTRUCTORS;
+            break;
+          case FeedbackVisibilityType.STUDENTS:
+            return CommentVisibilityType.STUDENTS;
+            break;
+          case FeedbackVisibilityType.RECIPIENT:
+            return CommentVisibilityType.RECIPIENT;
+            break;
+          case FeedbackVisibilityType.GIVER_TEAM_MEMBERS:
+            return CommentVisibilityType.GIVER_TEAM_MEMBERS;
+            break;
+          case FeedbackVisibilityType.RECIPIENT_TEAM_MEMBERS:
+            return CommentVisibilityType.RECIPIENT_TEAM_MEMBERS;
+            break;
+          default:
+            return CommentVisibilityType.GIVER;
+        }
+      });
+    this.showResponsesToInCommentVisibilityType.push(CommentVisibilityType.GIVER);
+
+    // push the new comment edit form
+    this.commentsTableModel.push({
+      commentEditFormModel: {
+        commentText: '',
+        commentVisibility:
+            this.getCommentVisibilityMap(this.showResponsesToInCommentVisibilityType,
+                this.showResponsesToInCommentVisibilityType),
+      },
+      timezone: this.timeZone,
+      isEditing: true,
+    });
+
+    this.comments.forEach((comment: FeedbackResponseComment) => {
+      this.commentsTableModel.push({
+        originalComment: comment,
+        timezone: this.timeZone,
+        commentEditFormModel: {
+          commentText: comment.commentText,
+          commentVisibility: this.getCommentVisibilityMap(comment.showCommentTo, comment.showGiverNameTo),
+        },
+        isEditing: false,
+      });
+    });
+  }
+
+  /**
+   * Triggers the delete comment event.
+   */
+  triggerDeleteCommentEvent(commentId: number): void {
+    this.commentService.deleteComment(commentId, Intent.INSTRUCTOR_RESULT).subscribe(() => {
+      this.commentsTableModel = this.commentsTableModel.filter((commentRow: CommentRowModel) =>
+          !commentRow.originalComment || commentRow.originalComment.feedbackResponseCommentId !== commentId);
+      const updatedComments: FeedbackResponseComment[] =
+          this.comments.filter((comment: FeedbackResponseComment) => comment.feedbackResponseCommentId !== commentId);
+      this.commentsChange.emit(updatedComments);
+    });
+  }
+
+  /**
+   * Triggers the update comment event.
+   */
+  triggerUpdateCommentEvent(index: number): void {
+    const commentData: CommentRowModel = this.commentsTableModel[index];
+
+    this.commentService.updateComment({
```

Comment:
It is not a good practice to fire API call in a component. But let's leave it first in case there are changes to the result page.